set-dark-theme-variant.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+# Based on code by Stephan Sokolow
+# Source: https://gist.github.com/ssokolow/e7c9aae63fb7973e4d64cff969a78ae8
+
+from contextlib import contextmanager
+import Xlib
+import Xlib.display
+import time
+import subprocess
+
+# Connect to the X server and get the root window
+disp = Xlib.display.Display()
+root = disp.screen().root
+
+# Prepare the property names we use so they can be fed into X11 APIs
+NET_ACTIVE_WINDOW = disp.intern_atom('_NET_ACTIVE_WINDOW')
+NET_WM_NAME = disp.intern_atom('_NET_WM_NAME')  # UTF-8
+WM_NAME = disp.intern_atom('WM_NAME')           # Legacy encoding
+last_seen = { 'xid': None, 'title': None }
+
+@contextmanager
+def window_obj(win_id):
+    """Simplify dealing with BadWindow (make it either valid or None)"""
```

Comment:
Maybe you can remove this context manager and just do:

```python
window_obj = None
with contextlib.suppress(Xlib.error.XError):
	window_obj = disp.create_resource_object('window', win_id)
```

Reply:
Nope. Getting rid of the context manager and just doing:

```python3
window_obj = None
with contextlib.suppress(Xlib.error.XError):
	window_obj = disp.create_resource_object('window', win_id)
return window_obj```

Breaks everything. If I had to guess it's because internally window is a connection to a window not really a window itself.

Reply:
Not sure where my comment went... but, getting rid of the context manager breaks everything. From what I can tell the "window" is like a socket or maybe it is a socket? I'm not sure how exactly X works? But just grabbing a reference and setting a prop doesn't work.